### ISSUE TITLE
[Feature Request]: The ability to connect to Google Drive and, if possible, to iCloud

### ISSUE THREAD

**Issue by MartinJDavis93** [sboesen/remotely-secure#35]
### What feature are you suggesting?

I think it's obvious that many would like to have the ability to connect to Google Drive and possibly to iCloud (though I'm not sure how feasible or whether Apple even provides such an option). Therefore, in the future, after all the planned changes you have, I hope you can consider implementing such integration.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by sboesen** [comment]
Definitely want to add support for Google Drive - iCloud I am also not sure the feasibility. Will need to investigate and figure out the effort required to add these. Currently prioritizing fixing/improving the webdav implementation as it appears buggy in some cases, #16. Once we have that and other pressing bugs ironed out definitely willing to explore adding new integrations.

Thanks!

**Comment by MartinJDavis93** [comment]
> Definitely want to add support for Google Drive - iCloud I am also not sure the feasibility. Will need to investigate and figure out the effort required to add these. Currently prioritizing fixing/improving the webdav implementation as it appears buggy in some cases, #16. Once we have that and other pressing bugs ironed out definitely willing to explore adding new integrations.
> 
> Thanks!

Yes, certainly, this is already a request for the future, after the main work is done. In general, the more storage options there are, the more support issues there may be since they all have different structures. And that can be challenging. However, it would be great to have all the major services: Google Drive, OneDrive, iCloud, Dropbox, Amazon, NextCloud, which the vast majority of users use. Because in [Remotely Save](https://github.com/remotely-save/remotely-save/blob/master/docs/services_connectable_or_not.md), they considered many others, but if managed to implement integration with these major ones, that would be incredible.

**Comment by sboesen** [comment]
100%, I want to support as many as possible and definitely agreed it adds code complexity/support work. I do want to prioritize sync options that users actively want because of the limited time I have to update the plug-in & the limited popularity of this plug-in (currently).

I will update the readme to call this out and ask users to open feature requests for those other sync options you mentioned if they want to use those options.

**Comment by MikeWi11s0n** [comment]
+1

**Comment by fyears** [comment]
(remotely-save author here) 

The major problem of GDrive support is that Google provides no obvious way to fire a request without server (Last time I checked). The plugin tries to keep working the same across mobile and computer, and also tries to be “serverless” to avoid many issues.

**Comment by sboesen** [comment]
I think it may be possible to add using [this library](https://googleapis.dev/nodejs/googleapis/latest/drive/index.html), but barely looked into it so far. Could definitely be a limitation preventing this!
